xarray/core/dataset.py

```diff
@@ -7155,13 +7155,12 @@ def _to_dataframe(self, ordered_dims: Mapping[Any, int]):
             k
             for k in self.variables
             if k not in self.dims
-            and not isinstance(self.variables[k].data, ExtensionDuckArray)
+            and not is_extension_array_dtype(self.variables[k].data)
```

Comment:
@TomNicholas see for example here about checking explicitly, so decoupling this PR should be doable.  Behavior on tests seemed the same